.github/workflows/website-deploy-preview.yml

```
@@ -0,0 +1,69 @@
+name: "website-deploy-preview"
+
+on:
+  workflow_dispatch:
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - "website/**"
+
+env:
+  AWS_REGION: us-east-2
+  IAM_ROLE_ARN: arn:aws:iam::068007702576:role/cplive-plat-ue2-dev-atmos-docs-github-action
+  IAM_ROLE_SESSION_NAME: cloudposse-atmos-ci-deploy-pr-${{ github.event.pull_request.number }}
+  S3_BUCKET_NAME: cplive-plat-ue2-dev-atmos-docs-origin
+  PR_NUMBER: ${{ github.event.pull_request.number }}
+
+# These permissions are needed to interact with GitHub's OIDC Token endpoint
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  website-deploy-preview:
+    runs-on: ubuntu-latest
+    steps:
+      # https://github.com/marketplace/actions/configure-aws-credentials-action-for-github-actions
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ env.IAM_ROLE_ARN }}
+          role-session-name: ${{ env.IAM_ROLE_SESSION_NAME }}
+
+      - name: Setup Node.js 16.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
```

If you use a `.nvmrc` file with `v16` you can set this line to

```suggestion
          node-version-file: '.nvmrc'
```

---

never used .nvmrc file, what the purpose of it and what issues it will solve for us? How it will be different from not using it? @nitrocode

---

It's for setting a default node version via `npm use`. This will have 2 benefits. One, we will be able to codify the stable node version used to generate the docs and two, the github workflow(s) can reuse the same file without having to put the version in multiple places.

---

added the file, it's workin ok, thanks